src/main/plugins/org.dita.xhtml/xsl/xslhtml/troubleshootingdisplay.xsl

```diff
@@ -0,0 +1,162 @@
+        <xsl:with-param name="id" select="'trouble_condition'"/>
+      </xsl:call-template>
+    </xsl:with-param>
+  </xsl:apply-templates>
+</xsl:template>
+
+  <xsl:template match="*[contains(@class,' troubleshooting/cause ')]" mode="dita2html:section-heading">
+  <xsl:apply-templates select="." mode="generate-task-label">
+    <xsl:with-param name="use-label">
+      <xsl:call-template name="getVariable">
+        <xsl:with-param name="id" select="'trouble_cause'"/>
+      </xsl:call-template>
+    </xsl:with-param>
+  </xsl:apply-templates>
+</xsl:template>
+  
+  <xsl:template match="*[contains(@class,' troubleshooting/remedy ')]" mode="dita2html:section-heading">
+    <xsl:apply-templates select="." mode="generate-task-label">
+      <xsl:with-param name="use-label">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'trouble_remedy'"/>
+        </xsl:call-template>
+      </xsl:with-param>
+    </xsl:apply-templates>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' troubleshooting/responsibleParty ')]" mode="dita2html:section-heading">
+    <xsl:apply-templates select="." mode="generate-task-label">
+      <xsl:with-param name="use-label">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'trouble_responsibleParty'"/>
+        </xsl:call-template>
+      </xsl:with-param>
+    </xsl:apply-templates>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' task/tasktroubleshooting ')]" mode="dita2html:section-heading">
+    <xsl:apply-templates select="." mode="generate-task-label">
+      <xsl:with-param name="use-label">
+        <xsl:call-template name="getVariable">
+          <xsl:with-param name="id" select="'trouble_task'"/>
+        </xsl:call-template>
+      </xsl:with-param>
+    </xsl:apply-templates>
+  </xsl:template>
+
+<!-- 
+     To override the task label for a specific element, match that element with this mode. 
+     For example, you can turn off labels for <context> with this rule:
+     <xsl:template match="*[contains(@class,' task/context ')]" mode="generate-task-label"/>
+-->
+<xsl:template match="*" mode="generate-task-label">
+  <xsl:param name="use-label"/>
+  <xsl:if test="$GENERATE-TASK-LABELS='YES'">
+    <xsl:variable name="headLevel">
+      <xsl:variable name="headCount">
+        <xsl:value-of select="count(ancestor::*[contains(@class,' topic/topic ')])+1"/>
+      </xsl:variable>
+      <xsl:choose>
+        <xsl:when test="$headCount > 6">h6</xsl:when>
+        <xsl:otherwise>h<xsl:value-of select="$headCount"/></xsl:otherwise>
+      </xsl:choose>
+    </xsl:variable>
+    <div class="troubleshootinglabel">
+      <xsl:element name="{$headLevel}">
+        <xsl:attribute name="class">sectiontitle troubleshootinglabel</xsl:attribute>
+        <xsl:value-of select="$use-label"/>
+      </xsl:element>
+    </div>
+  </xsl:if>
+</xsl:template>
+
+  <!-- Tasks have their own group. -->
```

Comment:
I think the rest of this can be removed - we should not be modifying task links within the troubleshooting override. I'm not sure if you intended to create special processing for troubleshooting links.